mlflow/utils/environment.py

```diff
@@ -729,6 +731,122 @@ def _get_pip_install_mlflow():
+    file_path: str,
+) -> None:
+    """
+    Removes requirements from the given requirements file (either `conda.yaml`
+    or `requirements.txt`). If any of the provided requirements are not present
+    in the file, they will be ignored. This function also ignores version specifiers.
+    For example, if the file contains "pandas==1.2.3", and the provided requirement
+    is "pandas", the function will remove "pandas==1.2.3" from the file.
+
+    Args:
+        reqs_to_remove (List[`Requirement`]): List of requirements to remove from the file
+        file_path (str): Path to the locally-saved requirements file. Must point to
+            either a `conda.yaml` or `requirements.txt` file.
+    """
+    old_reqs = _get_requirements_from_file(file_path)
+    updated_reqs = _remove_requirements(reqs_to_remove, old_reqs)
+    _write_requirements_to_file(file_path, updated_reqs)
+    _logger.info(
+        "Done removing requirements!\n\n"
+        f"Old requirements:\n{pformat([str(req) for req in old_reqs])}\n\n"
+        f"Updated requirements:\n{pformat(updated_reqs)}\n"
+    )
+
+
+def _get_requirements_from_file(
+    file_path: str,
+) -> List[Requirement]:
+    file_name = os.path.basename(file_path)
+    if file_name == _CONDA_ENV_FILE_NAME:
+        with open(file_path) as file:
+            conda_env = yaml.safe_load(file)
+
+        return [Requirement(s) for s in _get_pip_deps(conda_env)]
+    elif file_name == _REQUIREMENTS_FILE_NAME:
+        with open(file_path) as file:
+            reqs = file.readlines()
+
+        return [Requirement(req) for req in reqs]
+    else:
+        raise MlflowException.invalid_parameter_value(
+            "Invalid file path provided to _get_requirements_from_file! "
+            "Must be a path to either a `conda.yaml` or `requirements.txt` file. "
+            f"Received file path: {file_path}"
+        )
+
+
+def _write_requirements_to_file(
+    file_path: str,
+    new_reqs: List[str],
+) -> None:
+    file_name = os.path.basename(file_path)
+    if file_name == _CONDA_ENV_FILE_NAME:
+        with open(file_path) as file:
+            conda_env = yaml.safe_load(file)
+        conda_env = _overwrite_pip_deps(conda_env, new_reqs)
+        with open(file_path, "w") as file:
+            yaml.dump(conda_env, file)
+    elif file_name == _REQUIREMENTS_FILE_NAME:
+        with open(file_path, "w") as file:
+            file.write("\n".join(new_reqs))
+    else:
+        raise MlflowException.invalid_parameter_value(
+            "Invalid file path provided to _write_requirements_to_file! "
+            "Must be a path to either a `conda.yaml` or `requirements.txt` file. "
+            f"Received file path: {file_path}"
+        )
+
+
+def _add_or_overwrite_requirements(
+    new_reqs: List[Requirement],
+    old_reqs: List[Requirement],
+) -> List[str]:
+    old_reqs_dict = {req.name: str(req) for req in old_reqs}
+    new_reqs_dict = {req.name: str(req) for req in new_reqs}
```

Comment:
For the new input dicts, should we validate the inputs here? We perform validation of the requirements correctness between inferred and overrides when serializing the model instances by calling `_deduplicate_requirements()` to ensure that a submitting list of requirements overrides won't be unable to be installed. 
Since this API is effectively capable of 'breaking' a model, we should validate that a user isn't about to make their model incapable of being served.

Reply:
oh true, i'll call `_deduplicate_requirements()` on the resulting combined list just to make sure that the end result is valid.

Reply:
oh wait nevermind, the way it's implemented here should prevent most of these bad cases from occuring in the resulting list, since it's a dict keyed on package name. it should be sufficient to call it on `new_reqs`.